### PR TITLE
Adds ability to specify pdftotext options

### PR DESCRIPTION
**Pull request by philgooch**
* Adds test for `-table` option that produces tabular output

### CODE REVIEW

**Comment by jonmagic** [comment]
@philgooch thank you for the PR! I still need to pull it down and try it out but excited to see this :metal: To be future proof I think I'd like to see options be a hash instead of an array, maybe with a `flags` key which takes the array of flags. Thoughts on that approach?

**Comment by philgooch** [comment]
@jonmagic good idea, I've just pushed an update.  One thing I've found is that the `pdftotext` linked to by `xpdf` is superior to the one linked to by `poppler` in that it handles end-of-line breaks in the pdf in a more natural way (it wraps them rather than making them hard breaks), and also has the `-tables` option (`poppler` has just `-layout` which is not as accurate for tabular data)

File: lib/grim/page.rb [hunk 1]
```diff
@@ -47,11 +47,11 @@ def save(path, options={})
     #   pdf[1].text
     #   # => "This is text from slide 2.\n\nAnd even more text from slide 2."
     #
-    #   pdf[1].text(options=["-table"])
+    #   pdf[1].text(options={flags: ["-table"]})
```
Comment (jonmagic) [inline lib/grim/page.rb]: I think this should just be:

```ruby
#   pdf[1].text({flags: ["-table"]})
```

File: lib/grim/page.rb [hunk 2]
```diff
@@ -47,11 +47,11 @@ def save(path, options={})
...
     # Returns a String.
     #
-    def text(options=[])
-      command = [@pdftotext_path, "-enc", "UTF-8", "-f", @number, "-l", @number, options.join(", "), Shellwords.escape(@pdf.path), "-"].join(' ')
+    def text(options={flags: []})
```
Comment (jonmagic) [inline lib/grim/page.rb]: I think this should change to something more like:

```ruby
def text(options={})
  flags = options.fetch(:flags, [])
  command_parts = [@pdftotext_path, "-enc", "UTF-8", "-f", @number, "-l", @number]
  command_parts += flags if flags.length > 0
  command_parts += [Shellwords.escape(@pdf.path), "-"]
  command = command_parts.join(' ')
```

File: spec/lib/grim/page_spec.rb [hunk 3]
```diff
@@ -53,7 +53,7 @@
 
     it "should extract tabular data with the -table option" do
       pdf = Grim::Pdf.new(fixture_path("table.pdf"))
-      expect(pdf[0].text(options=["-table"])).to \
+      expect(pdf[0].text(options={flags: ["-table"]})).to \
```
Comment (jonmagic) [inline spec/lib/grim/page_spec.rb]: Can drop the `options=` here similar to the comment above.

**Comment by philgooch** [comment]
@jonmagic thanks for reviewing the PR, changes made as requested 😸

**Comment by jonmagic** [comment]
Published https://rubygems.org/gems/grim/versions/1.3.1

**Pull request merged**